src/Avalonia.Input/ContentInputElement.cs

```diff
@@ -0,0 +1,787 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using Avalonia.Controls;
+using Avalonia.Controls.Metadata;
+using Avalonia.Input.GestureRecognizers;
+using Avalonia.Input.TextInput;
+using Avalonia.Interactivity;
+using Avalonia.VisualTree;
+
+namespace Avalonia.Input
+{
+    [PseudoClasses(":disabled", ":focus", ":focus-visible", ":pointerover")]
+    public abstract class ContentInputElement : StyledElement, IInputElement
```

Comment:
There is some code duplication here with InputElement class.
Was it considered to move some implementation to the shared class with composition usage here? It could be internal or public if it should include dependency properties/events.

Or/and maybe can we change order of inheritance so ```Interactive``` class won't inherit ```Visual```?
```AvaloniaObject -> Animatable -> StyledElement -> Visual -> Layoutable -> Interactive -> InputElement```
to 
```AvaloniaObject -> Animatable -> StyledElement -> Interactive -> Visual -> Layoutable -> InputElement```
and
```AvaloniaObject -> Animatable -> StyledElement -> Interactive -> ContentInputElement```

@grokys

Reply:
It seems we don't have any other ```Visual```/```Layoutable``` in the codebase that doesn't inherit ```Interactive``` (and even ```InputElement```).
Is it theoretically possible to have ```Visual``` without input/interactions?

Reply:
There are things in flow layout that are only visual but not input elements (UIElementContainer). We don't know yet if those will _have to_ stay that way or not, so I am hesistent to change it.

I'll have another stab at extracting common functionality however.

Reply:
@maxkatz6 I tried at least moving out the base-implementation of Interactive (into EventHandlers).
For the IInputElement implementation, I don't know how to actually make it better. The properties are currently duplicated, but if I make them attached, I'd still have to give them an owner class.
On top of that, most of the code is virtual methods, event implementations, etc. without too much actual implementation code.